tests/functional/tests/file_manager/suite_file_manager.cpp

```diff
@@ -0,0 +1,78 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstddef>
+
+#include "FATFileSystem.h"
+#include "FileManagerKit.h"
+#include "SDBlockDevice.h"
+#include "boost/ut.hpp"
+#include "filesystem"
+#include "tests/config.h"
+#include "tests/ft_utils.h"
+
+using namespace leka;
+using namespace boost::ut;
+
+SDBlockDevice sd_blockdevice(SD_SPI_MOSI, SD_SPI_MISO, SD_SPI_SCK);
+FATFileSystem fatfs("fs");
+
+void initializeSD()
+{
+	constexpr auto default_sd_blockdevice_frequency = uint64_t {25'000'000};
+
+	sd_blockdevice.init();
+	sd_blockdevice.frequency(default_sd_blockdevice_frequency);
+
+	fatfs.mount(&sd_blockdevice);
+}
+
+suite suite_file_manager_kit = [] {
+	auto file = FileManagerKit::File {};
+
+	const std::filesystem::path tmp_dir			  = "/fs/var/tmp";
+	const std::filesystem::path test_dir		  = tmp_dir / "test/";
+	const std::filesystem::path test_sub_dir	  = test_dir / "sub/";
+	const std::filesystem::path test_filename	  = test_dir / "test_spike_lk_filesystem_kit";
+	const std::filesystem::path test_sub_filename = test_sub_dir / "test_spike_lk_filesystem_kit";
+	auto input_data								  = std::string {"Hello Leka"};
+
+	initializeSD();
+
+	std::filesystem::remove(test_filename);
+	std::filesystem::remove(test_sub_dir);
+	std::filesystem::remove(test_dir);
+
+	"open existing file"_test = [&] {
+		auto open = file.open("/fs/usr/test/file-1-ok.txt", "w");
+
+		expect(open >> fatal) << "Failed to open file";
+	};
+
+	"create directory"_test = [&] {
+		auto created = FileManagerKit::create_directory(test_dir);
+
+		expect(created >> fatal) << "Failed to create directory" << test_dir;
+	};
+
+	"create subdirectory"_test = [&] {
+		auto created = FileManagerKit::create_directory(test_sub_dir);
+
+		expect(created >> fatal) << "Failed to create directory";
+	};
+
+	"open new file"_test = [&] {
+		auto open = file.open(test_filename, "w");
+
+		expect(open >> fatal) << "Failed to open file";
+	};
+
+	"write to new file "_test = [&] {
+		auto bytes = file.write(input_data);
+
+		expect(bytes == std::size(input_data) >> fatal) << "Failed to write file";
+	};
+
+	file.close();
```

Comment:
I would remove the directories+files again here, so you know you leave the fs in a sane state.